apps/ewallet/lib/ewallet/gates/reset_password_gate.ex

```diff
@@ -0,0 +1,69 @@
+defmodule EWallet.ResetPasswordGate do
+  @moduledoc """
+  Handles a user's password reset.
+  """
+  alias EWalletDB.{ForgetPasswordRequest, User}
+
+  @doc """
+  Creates a reset password reset.
+  """
+  @spec request(map()) :: %ForgetPasswordRequest{}
+  def request(email) do
+    with {:ok, user} <- get_user_by_email(email),
+         {_, _} <- ForgetPasswordRequest.disable_all_for(user),
```

Comment:
Now I'm also wondering about the `disable_all`. Someone could spam reset an email address, preventing a real user from doing a password reset. I'm thinking we can add a lifetime to requests instead? We could add a setting for the request lifetime.